### ISSUE TITLE
Sequence function

### ISSUE THREAD

**Issue by shinriyo** [dentedpixel/LeanTween#25]
Will you add the sequence function?
Other tween tool always have the function.
I hope the function, or do you have already added?

**Comment by jjhesk** [comment]
+1

**Comment by superprat** [comment]
Sequencing function would help when trying to apply multiple tweens to the same object from different threads. Struggling with this issue right now. Tried calling LeanTween.cancel on the gameObject but that obviously doesn't finish the execution of the tween.

**Comment by dentedpixel** [comment]
You can call LeanTween.cancel( gameObject, doesComplete) with a doesComplete = yes which will call any onComplete methods on the cancelled tweens. Or were you thinking you would want all the tweens to get to their end positions as well? Let me think about the sequencer feature again, since it seems to be an often requested add on...

**Comment by superprat** [comment]
Yes, was looking to have them get to their end positions as well.

Thinking of building a timeline helper class that would sequence the tweens
for gameobjects

**Comment by dentedpixel** [comment]
Ah ok, yeah the LeanTween Visual Editor works a lot like a timeline editor, if you want to give that a shot. Or you could build your own, that would be cool too.

You can do a bit of a sequencer like steps by using the delayedCall method in a manner like:

int delayedId = LeanTween.delayedCall(gameObject, 4f, ()=>{
	LeanTween.moveY(gameObject, 2f, 3f).setFrom(1f));
	LeanTween.scaleX(gameObject, 0f, 1f).setDelay(3f).setFrom(2f);
}).setRepeat(3).setOnCompleteOnStart(true);

// has that sequence of moveY and scaleX repeat 3 times

**Comment by superprat** [comment]
I meant a Timeline editor to manage sequential tweens kind of like the greensock Timeline Library(http://greensock.com/timelinemax) it manages the sequence of tweens and their execution.

The problem I'm facing right now is that once a tween starts executing and external input adds another tween to the same gameObject, There's no way for me to sequence that on top of the previous tween. If the LeanTween.cancel( gameObject, doesComplete) would complete the running tween it would solve the problem.

But I think long term it would benefit a lot from having a sequencer and the flexibility to manage those things.

**Comment by neonwarge04** [comment]
I really wonder why this library doesn't contain sequencing function. Every developer with sane mind would not even forget it, it will be on the top of their TODO list. Unbelievable, lackluster approach on developing the API. 

For example, as suggested here, use onComplete. Here is what my code looks like. It looks like it has been coded by 5 year old, such a mess:

`		LeanTween
			.scale (goldCollectedText.gameObject, new Vector3 (1f, 1f, 1f), 0.15f)
			.setOnComplete (() => 
				{
					LeanTween
						.scale(goldCollectedValue.gameObject, new Vector3(1f, 1f, 1f), 0.15f)
						.setOnComplete(() =>
							{
								LeanTween
									.scale(highestGoldCollectedText.gameObject, new Vector3(1f, 1f, 1f), 0.15f)
									.setOnComplete(()=>
										{
											LeanTween
												.scale(highestGoldCollectedValue.gameObject, new Vector3(1f, 1f, 1f), 0.15f)
												.setOnComplete(OnGameOverButtonsEnter)
												.setEase(LeanTweenType.easeOutBounce);
										})
									.setEase(LeanTweenType.easeOutBounce);
							})
						.setEase(LeanTweenType.easeOutBounce);
				})
			.setEase (LeanTweenType.easeOutBounce);
`

Can't believe I have to write my own sequence method to get this done. Unbelievable. ADD IT! Quickly!

**Comment by AlmightyR** [comment]
@neonwarge04 A sequencer is just syntax sugar... So no, @dentedpixel didn't forget it. And no, it shouldn't be the top of his TODO list. And his effort regarding LeanTween is certainly not "lackluster"; it's one of the, if not *the* API with the best syntax-design and optimization.

If your code looks like that, it's because you've made it like that. Creating your own sequencer for LeanTween is not hard, and even if that was the case, neither is it hard to implement the same thing you did, in a more sane manner, instead of making that mess you've made.

Yes, sequencing functionality would be welcome, but no, it's not something of which the API needs. @dentedpixel is welcome to implement it (or not) at his discretion, but he should have no obligation to do your syntax-sugaring for you.

**Comment by dentedpixel** [comment]
I have started a sequencer, please check out the sequencer example scene: GeneralSequences

Right now it's pretty basic, no ability to play a sequence backwards or anything fancy like that.

Also before releasing I will make sure it's done in a way that doesn't effect garbage collection.

**Comment by TurkerTunali** [comment]
Thank you for your efforts.
